fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -527,7 +528,7 @@ public byte[] readStoredFields(String segmentName, int docId) throws IOException
         final byte[] key = storedFieldsSubspace.pack(Tuple.from(segmentName, docId));
         final byte[] rawBytes = context.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_GET_STORED_FIELDS,
                 context.instrument(LuceneEvents.Events.LUCENE_READ_STORED_FIELDS,
-                        context.ensureActive().get(key)));
+                        agilityContext.get(key)));
```

Comment:
Why this change? Is this something that needs to be applied over all the transaction access in the directory?

Reply:
Had a discussion with @jjezra and it looks like this should always the case - maybe the original Context should be removed?